HIPSYCL_PLATFORM not settable in CMakeLists.txt

Setting `-DHIPSYCL_PLATFORM=cpu` on the command line works fine but using `set(HIPSYCL_PLATFORM "cpu")` in the CMakeLists.txt file gives the following error:
```CMake Error at /opt/hipSYCL/lib/cmake/CMakeSYCLInformation.cmake:53 (message):
  More than one hipSYCL target platform is available.

  Please specify HIPSYCL_PLATFORM=cpu|cuda|rocm
Call Stack (most recent call first):
  /opt/hipSYCL/lib/cmake/hipsycl-config.cmake:9 (enable_language)
  CMakeLists.txt:15 (find_package)
```
I could reproduce this issue using the [example](https://github.com/illuhad/hipSYCL/blob/master/doc/examples.md) from the docs. If required, I can provide the CMakeLists.txt used.

--- a/cmake/CMakeSYCLInformation.cmake
+++ b/cmake/CMakeSYCLInformation.cmake
@@ -39,7 +39,7 @@ if(HIPSYCL_ROCM_BACKEND_AVAILABLE)
   endif()
 endif()
 
-set(HIPSYCL_PLATFORM "" CACHE STRING "The platform that hipSYCL should target. One of ${HIPSYCL_PLATFORMS_STRING}.")
+set(HIPSYCL_PLATFORM "${HIPSYCL_PLATFORM}" CACHE STRING "The platform that hipSYCL should target. One of ${HIPSYCL_PLATFORMS_STRING}.")
 
 # If HIPSYCL_PLATFORM has not been explicitly set by the user, first try to find
 # the corresponding environment variable. If that isn't set either, and only

